internal/runners/builds/builds.go

```diff
@@ -0,0 +1,202 @@
+package builds
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runbits/rationalize"
+	"github.com/ActiveState/cli/internal/runbits/runtime"
+	bpModel "github.com/ActiveState/cli/pkg/platform/api/buildplanner/model"
+	auth "github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildplan"
+	"github.com/ActiveState/cli/pkg/platform/runtime/store"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+type primeable interface {
+	primer.Outputer
+	primer.Auther
+	primer.Projecter
+	primer.SvcModeler
+	primer.Configurer
+	primer.Analyticer
+}
+
+type Params struct {
+	All bool
+}
+
+type Builds struct {
+	out       output.Outputer
+	project   *project.Project
+	analytics analytics.Dispatcher
+	svcModel  *model.SvcModel
+	auth      *auth.Auth
+	config    *config.Instance
+}
+
+type StructuredOutput struct {
+	Platforms []*structuredPlatform `json:"platforms"`
+}
+
+func (o *StructuredOutput) MarshalStructured(output.Format) interface{} {
+	return o
+}
+
+type structuredPlatform struct {
+	ID       string             `json:"id"`
+	Name     string             `json:"name"`
+	Builds   []*structuredBuild `json:"builds"`
+	Packages []*structuredBuild `json:"packages"`
+}
+
+type structuredBuild struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+	URL  string `json:"url"`
+}
+
+func New(p primeable) *Builds {
+	return &Builds{
+		out:       p.Output(),
+		project:   p.Project(),
+		auth:      p.Auth(),
+		svcModel:  p.SvcModel(),
+		config:    p.Config(),
+		analytics: p.Analytics(),
+	}
+}
+
+func rationalizeError(err *error) {
+	switch {
+	case err == nil:
+		return
+	case errors.Is(*err, rationalize.ErrNoProject):
+		*err = errs.WrapUserFacing(*err,
+			locale.Tr("err_no_project"),
+			errs.SetInput())
+	}
+}
+
+func (b *Builds) Run(params *Params) (rerr error) {
+	defer rationalizeError(&rerr)
+
+	if b.project == nil {
```

Comment:
While working on the `state builds dl` command I've come to realize that requiring the project to be present on the system is a bit of an odd UX choice. We can source all of this information remotely. 

If a user wanted to download the windows installer they would first have to download and install the entire project to get the ID they are interested in and then download the actual artifact they are interested in.

Reply:
That's a good point! Wanna file a followup? I'd rather land this as an improvement than add to the scope at this stage.

Reply:
Sure, filed a follow-up here: https://activestatef.atlassian.net/browse/DX-2513.

I will continue on with `state builds dl` using a locally checked out project and it can be iterated on in the above story.